contracts/base/ATM/ATMToken.sol

```diff
@@ -0,0 +1,285 @@
+  *
+  *  @author develop@teller.finance
+ */
+
+contract ATMToken is ATMTokenInterface, ERC20Detailed, ERC20Mintable, ERC20Burnable {
+    /**
+      *  @notice ATMToken implements an ERC20 token with a supply cap and a vesting scheduling
+     */
+     using SafeMath for uint256;
+
+    /* Modifiers */
+    /**
+        @notice Checks if sender is owner
+        @dev Throws an error if the sender is not the owner
+     */
+    modifier onlyOwner() {
+         require(msg.sender == _owner, "CALLER_IS_NOT_OWNER");
+         _;
+     }
+
+    /* State Variables */
+    uint256 private _cap;
+    uint256 private _maxVestingsPerWallet;
+    address private _owner;
+
+    /* Structs */
+    struct VestingTokens {
+        address account;
+        uint256 amount;
+        uint256 start;
+        uint256 cliff;
+        uint256 deadline;
+    }
+
+    /* Mappings */
+    mapping (address => mapping(uint256 => VestingTokens)) private _vestingBalances; // Mapping user address to vestings id, which in turn is mapped to the VestingTokens struct
+    mapping (address => uint256) public vestingsCount;
+    mapping (address => uint256) public assignedTokens;
+
+    /* Constructor */
+    constructor (
+        string memory _name,
+        string memory _symbol,
+        uint8 _decimals,
+        uint256 cap,
+        uint256 maxVestingsPerWallet
+    ) 
+        ERC20Detailed(_name, _symbol, _decimals)
+        public
+    {
+        require(cap > 0, "CAP_CANNOT_BE_ZERO");
+        _cap = cap;
+        _maxVestingsPerWallet = maxVestingsPerWallet;
+        _owner = msg.sender;
+    }
+
+    /* Functions */
+    /**
+     * @notice Returns the cap on the token's total supply
+     * @return The supply capped amount
+     */
+    function cap() 
+        external
+        view
+        returns (uint256) 
+    {
+        return _cap;
+    }
+
+    /**
+     * @notice Sets a new cap on the token's total supply.
+     * @param newCap The new capped amount of tokens
+     */
+    function setCap(
+        uint256 newCap
+    ) 
+        external
+        onlyOwner()
+    {
+        _cap = newCap;
+        emit NewCap(_cap);
+    }
+
+    /**
+     * @notice Increase account supply of specified token amount
+     * @param account The account to mint tokens for
+     * @param amount The amount of tokens to mint
+     * @return true if successful
+     */
+    function mint(
+        address account,
+        uint256 amount
+    ) 
+        public
+        onlyOwner()
+        returns (bool)
+    {
+        require(account != address(0x0), "MINT_TO_ZERO_ADDRESS_NOT_ALLOWED");
+        _beforeTokenTransfer(address(0x0), account, amount);
+        _mint(account, amount);
+        return true;
+    }
+
+    /** @notice Creates `amount` tokens and assigns them to `account`, increasing
+     * the total supply.
+     *
+     * Includes a vesting period before address is allowed to use tokens
+     *
+     * Emits a {Transfer} event with `from` set to the zero address.
```

Comment:
Should we set this as the owner or the ATM contract itself?

Reply:
Set it as the ATM contract for now. Can change it to owner